infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -128,6 +129,10 @@ private long getStartingSegment() {
      * Create stream log directory if not exists
      */
     private void initStreamLogDirectory() {
+        File corfuDir = new File(logDir.getParent().toString());
+        if (!corfuDir.canWrite()) {
```

Comment:
I believe this only check file permisions, but doesn't reflect the filesystem mount option. 

Please use:
File f = new File("/home/maithem/CorfuDB/n1");
FileStore fs = Files.getFileStore(Paths.get("/home/maithem/CorfuDB/n1"));
fs.isReadOnly();

Reply:
Added this check in addition to corfu directory and log directory checks.